collector/pkg/metadata/kubernetes/pod_watch.go

```diff
@@ -53,9 +96,32 @@ func (m *podMap) delete(namespace string, name string) {
 	if ok {
 		delete(podInfoMap, name)
 	}
+	globalWorkload.delete(namespace, name)
```

Comment:
Due to the same reason as the `add` method, move globalWorkload.delete out of the current scope.